### PR TITLE
NATIVE-559: Locale override

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **NATIVE**.

### CODE REVIEW

File: build-configs/configs/malte/index.js [hunk 1]
```diff
@@ -3,10 +3,13 @@
...
 import type { BuildConfigType } from '../BuildConfigType'
 
+const APP_NAME = 'Malte'
+
 const MalteBuildConfig: BuildConfigType = {
-  appName: 'Malte',
+  appName: APP_NAME,
   lightTheme,
   darkTheme,
+  localesOverride: APP_NAME,
```
Comment (klinzo) [inline build-configs/configs/malte/index.js:12]: Not sure whether we should identify what locales to use with something better :thinking: Proposals are welcome :)

**Comment by klinzo** [comment]
The build fails because the file `malte-locales.json` does not exist yet obviously. Depending on when that locales are available, I'll comment the corresponding lines of code out or wait until then.

**Comment by maxammann** [comment]
Flow is failing:

```

Cannot resolve module `../../../locales/malte-locales.json`.

   5| import malteOverrideLocales from '../../../locales/malte-locales.json'
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

File: src/modules/i18n/__mocks__/locales.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (maxammann) [inline src/modules/i18n/__mocks__/locales.json:1]: Should be shorter for better readable snapshots.

File: src/modules/i18n/__tests__/__snapshots__/loadLocales.spec.js.snap [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    "events": Object {
+      "currentlyNoEvents": "ليست هناك فعاليات متوفرة حاليًا.",
+      "date": "التاريخ",
+      "events": "الفعاليات",
+      "location": "الموقع",
+    },
+  },
+  "de": Object {
+    "app": Object {
+      "metaDescription": "Integreat ist Ihr digitaler Guide für Deutschland. Finden Sie lokale Informationen, Veranstaltungen und Beratung. Immer aktuell und in Ihrer Sprache.",
```
Comment (maxammann) [inline src/modules/i18n/__tests__/__snapshots__/loadLocales.spec.js.snap:26]:
```suggestion
      "metaDescription": "Malte ist Ihr digitaler Guide für Deutschland. Finden Sie lokale Informationen, Veranstaltungen und Beratung. Immer aktuell und in Ihrer Sprache.",
```

Comment (maxammann) [reply src/modules/i18n/__tests__/__snapshots__/loadLocales.spec.js.snap:26]: Or just remove these lines as they make the snapshot seem weird.

File: src/modules/i18n/loadLocales.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      }
+    })
+    return transformedLocales
+  },
+  {}
+)
+
+const loadLocales = (): TransformedLocalesType => {
+  const localesOverride = buildConfig().localesOverride
+  const locales = localesOverride ? merge(defaultLocales, localesOverride) : defaultLocales
```
Comment (maxammann) [inline src/modules/i18n/loadLocales.js:31]: Maybe add a check here like mentioned in the WEBAPP PR.

**Review by maxammann — APPROVED** [review]

**Comment by klinzo** [comment]
> Flow is failing:
> 
> ```
> 
> Cannot resolve module `../../../locales/malte-locales.json`.
> 
>    5| import malteOverrideLocales from '../../../locales/malte-locales.json'
>                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ```

See comment below. The json does not exist yet because the locales PR is not yet merged.

**Comment by maxammann** [comment]
@klinzo lets comment the code out then until we are done with the translations. We can remove the comment in the Malteser locales task and create a new PR

File: src/modules/i18n/loadLocales.js (hunk 4 above)
Comment (klinzo) [reply src/modules/i18n/loadLocales.js:31]: See discussion https://github.com/Integreat/integreat-webapp/pull/390#discussion_r475522181

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
I accidentally redid this in #307 

I added a more extensive mock and fixed a wrong unit test there.

**Comment by klinzo** [comment]
Lets merge your PR. More up to date, better mocks and unit tests :+1:

**Pull request closed without merging**